TypeCobol/Compiler/Preprocessor/ReplaceTokensLinesIterator.cs

```diff
@@ -427,31 +428,17 @@ private Token CreateReplacedTokens(Token originalToken, ReplaceOperation replace
                 // One pure partial word => one replacement token
                 case ReplaceOperationType.PartialWord:
                     PartialWordReplaceOperation partialWordReplaceOperation = (PartialWordReplaceOperation)replaceOperation;
-                    string tokenText = originalToken.Text;
-                    string partToReplace = partialWordReplaceOperation.ComparisonToken.Text;
+                    string normalizedTokenText = originalToken.NormalizedText;
+                    string normalizedPartToReplace = partialWordReplaceOperation.ComparisonToken.NormalizedText;
                     //#258 - PartialReplacementToken can be null. In this case, we consider that it's an empty replacement
                     var replacementPart = partialWordReplaceOperation.PartialReplacementToken != null ? partialWordReplaceOperation.PartialReplacementToken.Text : "";
-                    //replace every occurrences in the original token text
-                    int startIndex = 0;
-                    int indexOfPartToReplace;
-                    while (startIndex < tokenText.Length
-                           &&
-                           (indexOfPartToReplace = tokenText.IndexOf(partToReplace, startIndex, StringComparison.OrdinalIgnoreCase)) >= 0)
-                    {
-                        tokenText =
-                            (indexOfPartToReplace > 0
-                                ? tokenText.Substring(0, indexOfPartToReplace)
-                                : string.Empty) +
-                            replacementPart +
-                            (indexOfPartToReplace + partToReplace.Length < tokenText.Length
-                                ? tokenText.Substring(indexOfPartToReplace + partToReplace.Length)
-                                : string.Empty);
-                        startIndex = indexOfPartToReplace + replacementPart.Length;
-                    }
+
                     // TO DO use Replace(String, String, StringComparison) overload, available in .NET 5
```

Comment:
This TODO can be removed now ?

Reply:
![300px-Well_Yes,_But_Actually_No](https://user-images.githubusercontent.com/51322634/97448526-0604c580-1931-11eb-8d82-9c9d505afa42.jpg)
Here I use the `Regex.Replace` for the ignore case option but this is not a regular expression replace because `normalizedPartToReplace` is not a pattern but a plain string. `string.Replace` with StringComparison argument (introduced with .NET 5) is more appropriate.

Reply:
Ok ;)